iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -522,19 +519,23 @@ private async Task DisposeMessageAsync(string lockToken, AmqpIoTDisposeActions o
 
         private void TwinMessageListener(Twin twin, string correlationId, TwinCollection twinCollection)
         {
-            if (correlationId != null)
+            if (correlationId == null)
             {
-                // It is a GET, just complete the task.
-                TaskCompletionSource<Twin> task;
-                if (_twinResponseCompletions.TryRemove(correlationId, out task))
-                {
-                    task.SetResult(twin);
-                }
+                // This is desired property updates, so call the callback with TwinCollection.
+                _onDesiredStatePatchListener(twinCollection);
             }
             else
             {
-                // It is a PATCH, just call the callback with the TwinCollection
-                _onDesiredStatePatchListener(twinCollection);
+                if(correlationId.StartsWith(AmqpTwinMessageType.Get.ToString(), StringComparison.OrdinalIgnoreCase) ||
+                    correlationId.StartsWith(AmqpTwinMessageType.Patch.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
do we get a twin for the Patch operation as well, or is it null? how are the two cases handled when this task completes?

Reply:
This is just for completing a task which we stored in dictionary. The comment earlier was misleading.